pyEpiabm/pyEpiabm/sweep/spatial_sweep.py

```diff
@@ -118,13 +123,31 @@ def __call__(self, time: float):
                     max_weight = np.nanmax(distance_weights)
                     distance_weights = np.nan_to_num(distance_weights,
                                                      nan=max_weight)
-                cell_list = random.choices(possible_infectee_cells,
-                                           weights=distance_weights,
-                                           k=number_to_infect)
+                # Use of the cutoff distance idea from CovidSim.
+                cutoff = Parameters.instance().infection_radius
+                distance_weights = [weight if (cutoff > 1/weight) else 0 for
+                                    weight in distance_weights]
+                # Will catch the case if distance weights isn't configured
+                # correctly and returns the wrong length.
+                assert len(distance_weights) == len(possible_infectee_cells), (
+                    "Distance weights are not the same length as cell list")
+
+                try:
+                    # Will catch a list of zeros
+                    1 / sum(distance_weights)
```

Comment:
Are we using this result? If not just check the sum of the list is non-zero (copy the source code from the random module)

Reply:
I agree the source code seems to produce ValueErrors as described, but not able to get any to raise on my machine

Reply:
Written it in manually but not the slickest